vm.go

```diff
@@ -648,6 +670,9 @@ func (vm *vm) run() {
 				atomic.StoreUint32(&vm.interrupted, 0)
 			}()
 		} else {
+			if tickMetricsEnabled {
+				vm.trackMetrics(vm.prg)
+			}
```

Comment:
I'm just nervous about the performance implications of having this tick tracking here, I've added a few pieces that should mitigate. I think we could potentially also add sampling if we see performance is suffering in the benchmarks.

i.e. something like `if tickMetricsEnabled && ticks % 10 == 0`